lua/randomat2/events/sv_poker.lua

```diff
@@ -0,0 +1,1136 @@
+            if atLeastOne then
+                return true
+            else
+                atLeastOne = true
+            end
+        end
+    end
+
+    return false
+end
+
+local function CanDispenseWinnings()
+    local onePlayerStillAliveWithBets = false
+
+    for _, ply in ipairs(EVENT_REF.Players) do
+        if ply:Alive() and EVENT_REF.PlayerBets[ply] then
+            if onePlayerStillAliveWithBets then
+                return true
+            else
+                onePlayerStillAliveWithBets = true
+            end
+        end
+    end
+
+    return false
+end
+
+-- Called to register a player's bet (or lack thereof)
+function EVENT:RegisterPlayerBet(ply, bet, betAmount, forceBet)
+    if not self.Started then self:End() return end
+    if not ply:IsValid() then return end
+
+    -- If we receive a bet when we're not expecting (and it isn't a fold), ignore it
+    if not self.ExpectantBetter and bet > 1 and not forceBet then
+        return
+    end
+
+    if ply == self.ExpectantBetter or forceBet then
+        if ply == self.ExpectantBetter and timer.Exists("WaitingOnPlayerBet") then
+            timer.Remove("WaitingOnPlayerBet")
+        end
+
+        self.ExpectantBetter = nil
+        local highestBet = GetHighestBet()
+
+        if bet < BettingStatus.CHECK then
+            PlayerFolds(ply)
+        elseif bet == BettingStatus.CHECK then
+            if highestBet > betAmount then
+                PlayerFolds(ply)
+            else
+                PlayerChecks(ply)
+            end
+        elseif bet == BettingStatus.CALL then
+            PlayerCalls(ply)
+        elseif bet == BettingStatus.RAISE then
+            if betAmount <= highestBet then
+                PlayerCalls(ply)
+            else
+                PlayerRaises(ply, betAmount)
+            end
+        else
+            error(ply:Nick() .. " is sending net messages manually...")
+        end
+
+        if not forceBet then
+            if AllPlayersMatchingBets() then
+                net.Start("PlayersFinishedBetting")
+                net.Broadcast()
+
+                self:EndBetting()
+            else
+                local nextPly = GetNextValidPlayer(ply)
+                
+                self:BeginBetting(nextPly)
+            end
+        end
+    elseif bet < BettingStatus.CHECK then
+        -- Out of sync player fold, used primarily for player disconnecting/death
+        PlayerFolds(ply)
+
+        if not EnoughPlayersRemaining() then
+            if CanDispenseWinnings() then
+                self:CalculateWinner()
+            else
+                net.Start("DeclareNoWinner")
+                net.Broadcast()
+
+                timer.Simple(GetDynamicRoundTimerValue("RoundStateMessage"), function()
+                    self:End()
+                end)
+            end
+        end
+    end
+end
+
+function EVENT:EndBetting()
+    timer.Simple(GetDynamicRoundTimerValue("RoundStateMessage"), function()
```

Comment:
Recommend against using simple timers because they are not stoppable if, for instance, the event is ended early.